ext/test/cache_expiration_test.rb

```diff
@@ -0,0 +1,75 @@
+require 'net/http'
+require 'json'
+require 'rspec/expectations'
+require 'fileutils'
+
+include RSpec::Matchers
+
+class CacheTester
+  def initialize
+    @catalog         = 'testone'
+    @environment     = 'production'
+    @environment_uri = URI("http://localhost:8140/puppet/v3/catalog/#{@catalog}?environment=#{@environment}")
+    @cache_uri       = URI("http://localhost:8140/puppet-admin-api/v1/environment-cache")
+  end
+
+  def modify_environment
+    modules_path = File.join(__dir__, 'environments', @environment, 'modules')
+    tmp_path = File.join(__dir__, 'for-now')
+    FileUtils.mkdir_p(tmp_path)
+    FileUtils.mv(modules_path, tmp_path)
```

Comment:
Are you planning to then swap in different content?

Reply:
possibly? first I was just trying to see if it could continue to find these modules after I'd moved them, but that might be flawed approach..